src/pages/filter/ui/query-builder/query-builder-total-numbers.tsx

```diff
@@ -18,16 +18,24 @@ export const QueryBuilderTotalNumbers = observer((): ReactElement => {
     element => element.isActive || element.isReturnedVariantsActive,
   )
 
-  const returnedVariants = stepData[stepIndex]?.difference
-  const startVariants = stepData[stepIndex]?.startFilterCounts
-  const hasReturnedVariants = Boolean(returnedVariants)
-  const hasStartVariants = Boolean(startVariants)
+  const currentStep = stepData[stepIndex]
+  const difference = currentStep?.difference
+  const startFilterCounts = currentStep?.startFilterCounts
+  const hasReturnedVariants = Boolean(difference)
+  const hasStartVariants = Boolean(startFilterCounts)
+
+  const isFirstStepEmpty = stepData[0]?.groups.length === 0
+  const isTreeEmpty = stepData.length === 2 && isFirstStepEmpty
+  const shouldShowReturnedVariants = hasReturnedVariants && !isTreeEmpty
```

Comment:
if you interact with dtree store, maybe better to create getters in dtree like hasReturnedVariants and isTreeEmpty?

Reply:
Added getter "isTreeEmpty". It will not work to make a getter with "hasReturnedVariants ", since in order to get it we need to pass an index. Leave it as it is or make a function to get "hasReturnedVariants " ?

Reply:
leave as it is